### PR TITLE
[MLIR] Refactor FuncOpToLLVM to operate on FunctionOpInterface instead of FuncOp (NFC)

### PR DESCRIPTION
**Pull request by allatit23**
* refactor `convertFuncOpToLLVMFuncOp` to accept a `FunctionOpInterface` instead of func::FuncOp
* move `convertFuncOpToLLVMFuncOp` to corresponding public header, making it available for downstream project.

### CODE REVIEW

**Comment by allatit23** [comment]
@joker-eph Could you please take a look at this PR?

File: mlir/include/mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h [hunk 1]
```diff
@@ -14,13 +14,28 @@
...
+class LLVMFuncOp;
+} // namespace LLVM
+
+class ConversionPatternRewriter;
 class DialectRegistry;
 class LLVMTypeConverter;
 class RewritePatternSet;
 class SymbolTable;
 
+/// Convert input FuncOp to LLVMFuncOp by using the provided LLVMTypeConverter.
```
Comment (joker-eph) [inline mlir/include/mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h:32]:
```suggestion
/// Convert input FunctionOpInterface operation to LLVMFuncOp by using the provided LLVMTypeConverter.
```

File: mlir/include/mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h [hunk 2]
```diff
@@ -14,13 +14,28 @@
...
+/// Returns failure if failed to so.
```
Comment (joker-eph) [inline mlir/include/mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h:33]:
```suggestion
/// Convert input FunctionOpInterface operation to LLVMFuncOp by using the
/// provided LLVMTypeConverter. Return failure if failed to so.
```

**Comment by allatit23** [comment]
@joker-eph  Thanks. I do not have write access. Could you please help to merge it?

**Comment by joker-eph** [comment]
Your GitHub profile has a setting to make your email private, if I merge if it’ll show up as commit with 138413300+allatit23@users.noreply.github.com as authorship, I suspect we’d rather have a real email here

**Comment by allatit23** [comment]
@joker-eph I have made the email public. Could you please take a look again?

**Pull request merged**